### DISCUSSION TITLE
SRQ

### DISCUSSION THREAD

**Discussion by sansanda** [pymeasure/pymeasure#875] [Ideas]
Hi everybody.
Perhaps it is an stupid question but here these is.
I have to perform a task with an gpib instrument which takes long time. My first idea is to use the wait_for_srq of the Visa adapter. 

Whar do you think? It should be implemented in Instruments class?

Many thanks,

David

**Comment by BenediktBurger** [comment]
`wait_for_srq` seems a good idea (https://www.rohde-schwarz.com/us/driver-pages/remote-control/measurements-synchronization_231248.html), but I do not use gpib and cannot offer information based on experience.

I'd implement it in my code:
~~~python
inst = MyInstrument("gpibName")
inst.start_action()
inst.adapter.wait_for_srq()
inst.get_results()
~~~

**Comment by LongnoseRob** [comment]
@sansanda, it is not a stupid question.
There is one caveat, it does not work with all pyvisa-backends.
I think the commerical packages (NI-VISA,Keysight, R&S..) support it. However pyviosa-py does not support it,
At least that was the status a few month/years ago, when I looked into it for use uwith my Agilent 82357a & linux-gpib.
Also I should mention , I have not yet tested the commerical VISA packages under Linux yet..
